docs: user registration (#1925)
woodpecker-ci/woodpecker: docs/docs/30-administration/10-server-config.md

## User registration

Woodpecker does not have its own user registry; users are provided from your [forge](./11-forges/10-overview.md) (using OAuth2).

> Registration is closed by default (`WOODPECKER_OPEN=false`). If registration is open (`WOODPECKER_OPEN=true`) then every user with an account at the configured forge can login to Woodpecker.

Reviewer: ```suggestion
Registration is closed by default (`WOODPECKER_OPEN=false`). If registration is open (`WOODPECKER_OPEN=true`) every user with an account at the configured forge can login to Woodpecker.
```

    environment:
      - [...]
+     - WOODPECKER_OPEN=true
```

> When registration is closed, an administrator must add new users manually (with the CLI: `woodpecker-cli user add`) or by allowing specific users via `WOODPECKER_ADMIN` or `WOODPECKER_ORGS`.

Reviewer: or via ui 

maybe use a list to show options

To close registration, but allow specific users:
```diff
# docker-compose.yml
version: '3'

services:
  woodpecker-server:
    [...]
    environment:
      - [...]
+     - WOODPECKER_OPEN=false

> +     - WOODPECKER_ADMIN=johnsmith,janedoe

Reviewer: As the admin option is not related to open I would explain it without mentioning the open option

Reviewer: WOODPECKER_ADMIN is setting users to be admin not if they are allowed to login or not

Reviewer 2: https://github.com/woodpecker-ci/woodpecker/blob/a958049485d0284f48ed80e4650330ebe9b0aa65/server/api/login.go#L76

> WOODPECKER_ADMIN is setting users to be admin not if they are allowed to login or not

well yes but it's also used for that - to not have the chiken-egg problem

```

To close registration, but allow users who are members of approved organizations:
```diff
# docker-compose.yml
version: '3'

services:
  woodpecker-server:
    [...]
    environment:
      - [...]
+     - WOODPECKER_OPEN=false

> +     - WOODPECKER_ORGS=dolores,dogpatch

Reviewer: As the org option (afaik) is not related to open I would explain it without mentioning the open option

Reviewer: You could allow only the org of your company. If `WOODPECKER_OPEN=false` is set an admin would be needed to add all users by hand. In case of `WOODPECKER_OPEN=true` the whole company could register on their own

Reviewer 2: also WOODPECKER_OPEN has to be true if you wana use WOODPECKER_ORGS

https://github.com/woodpecker-ci/woodpecker/blob/a958049485d0284f48ed80e4650330ebe9b0aa65/server/api/login.go#L76-L91

    environment:
      - [...]
+     - WOODPECKER_OPEN=true
```

When registration is closed users can be added:
- By adding new users manually via the CLI: `woodpecker-cli user add`, or
- By allowing specific admin users via the `WOODPECKER_ADMIN` setting, or

> - By allowing specific orgaizations via the `WOODPECKER_ORGS` setting.

Reviewer 2: ```suggestion
- By allowing specific users via orgaizations membership via the `WOODPECKER_ORGS` setting.
```

> - By adding new users manually via the CLI: `woodpecker-cli user add`, or
> - By allowing specific admin users via the `WOODPECKER_ADMIN` setting, or
> - By allowing specific orgaizations via the `WOODPECKER_ORGS` setting.

Reviewer 2: ```suggestion
- by adding new users manually via the CLI: `woodpecker-cli user add`
- by allowing specific admin users via the `WOODPECKER_ADMIN` setting
- by allowing specific users via organization membership through the `WOODPECKER_ORGS` setting
```

To close registration, but allow specific admin users:
```diff
# docker-compose.yml
version: '3'

services:
  woodpecker-server:
    [...]
    environment:
      - [...]
+     - WOODPECKER_OPEN=false
+     - WOODPECKER_ADMIN=johnsmith,janedoe
```

> To close registration, but allow users who are members of approved organizations:

Reviewer 2: ```suggestion
To allow registration users, who are members of approved organizations:
```

```diff
# docker-compose.yml
version: '3'

services:
  woodpecker-server:
    [...]
    environment:
      - [...]

> +     - WOODPECKER_OPEN=false

Reviewer 2: ```suggestion
+     - WOODPECKER_OPEN=true
```
